### ISSUE TITLE
Fission site bounding box reporting when threshold fissionable site rejection sampling error is triggered

### ISSUE THREAD

**Issue by shikhar413** [openmc-dev/openmc#3073]
## Description

#2916 fixed an error check that was previously not properly implemented in `IndependentSource::sample,` where the variables `n_accept` and `n_reject` are now properly defined as static variables. Previously, only `n_reject` was defined as static, so the error check for checking the fraction of rejected particles was not being hit even though it should have been. What this means for some legacy models now is that if the independent source distribution bounding box was defined over a large domain of non-fissionable material, now the model needs to be updated to encompass a narrower domain that only spans the fissionable domain. The emergence of this error message is already happening for some Cardinal models through the version update in OpenMC (see https://github.com/neams-th-coe/cardinal/pull/903). This requires inspecting the geometry and trying to determine the minimum bounding box of fission sites, which can be quite tedious to do for more complicated models. Instead, what might be preferred here is if the error message for threshold percentage of rejected particles over fissionable areas is triggered, then OpenMC could also report the minimum bounding box from the fission sites that were samples so far

## Alternatives

None

## Compatibility

No changes to the core functionality, just more useful error reporting

**Comment by shikhar413** [comment]
Tagging @paulromano and @ebknudsen for their opinions on this

**Comment by shikhar413** [comment]
Also tagging @aprilnovak so she is aware of this issue

**Comment by ebknudsen** [comment]
In essence, I suppose this is simple enough to implement. One would simply keep a running log of the most extreme but accepted source site spatial coordinates. On the other hand, such corners would always be slightly inside a "fissionable-enclosing" bounding box, so depending on the use-case manual "fiddling" might be necessary anyway.

**Comment by shikhar413** [comment]
Yeah I think the idea here is that even if you have the fissionable enclosing box a bit smaller than the "ideal" size, if you run enough inactive cycles then it shouldn't affect your "converged" fission source distribution source distrubtion for use in the active cycles

**Comment by shikhar413** [comment]
The only thing I need to dig into is to figure out whether the sampled sites are defined for each MPI process, and whether some MPI communication is needed to get the bounding box defined over all MPI processes

**Comment by shikhar413** [comment]
That makes sense, that would be a lot more straightforward of a "fix".

By the way, when trying to run with turning the only fissionable constraints off and running in parallel, I see " WARNING: The shared fission bank is full. Additional fission sites created in this generation will not be banked. Results may be non-deterministic." warning on the first batch. Can this warning be ignored?
